model/device.go

```diff
@@ -78,6 +136,37 @@ func (da DeviceAttribute) Validate() error {
 	)
 }
 
+func (da DeviceAttribute) Equal(e DeviceAttribute) bool {
+	if (da.Value == nil || e.Value == nil) && !(da.Value == nil && e.Value == nil) {
+		return false
+	}
+	ok1 := reflect.TypeOf(da.Value).Kind() == reflect.Slice
+	ok2 := reflect.TypeOf(e.Value).Kind() == reflect.Slice
+	if ok1 || ok2 {
+		// there is no portable way to compare slices of unknown type
+		return false
+	} else {
+		if da.Value != e.Value {
+			return false
+		}
+	}
```

Comment:
> // there is no portable way to compare slices of unknown type

Yes, you can...
```suggestion
	rVal1 := reflect.ValueOf(v1)
	rVal2 := reflect.ValueOf(v2)
	if rVal1.Kind() != rVal2.Kind() {
		return false
	}
	if rVal1.Kind() == reflect.Slice {
		if rVal1.Len() != rVal2.Len() {
			return false
		}
		elemKind := rVal1.Elem().Kind()
		if elemKind != rVal2.Elem().Kind() {
			return false
		} else if elemKind != reflect.String && elemKind != reflect.Float64 {
			// Invalid DeviceAttribute
			return false
		}
		n := rVal1.Len()
		for i := 0; i < n; i++ {
			if !rVal1.Index(i).Equal(rVal2.Index(i)) {
				return false
			}
		}
	} else {
		if da.Value != e.Value {
			return false
		}
	}
```

Reply:
I tired that, but must have missed something, since ti failed for the primitive.A type. I think I missed the .Len() call.

Reply:
nice! thank you!